pkg/kubelet/container/helpers.go

```diff
@@ -87,7 +89,7 @@ func ShouldContainerBeRestarted(container *v1.Container, pod *v1.Pod, podStatus
 		return false
 	}
 	// Always restart container in the unknown, or in the created state.
-	if status.State == ContainerStateUnknown || status.State == ContainerStateCreated {
+	if status.State == ContainerStateUnknown || (!utilfeature.DefaultFeatureGate.Enabled(features.EventedPLEG) && status.State == ContainerStateCreated) {
```

Comment:
Thanks @pacoxu for fix. Would it make sense to add a comment on how checking for evented pleg feature fixes the issue? 

IMO, it might be helpful for anyone browsing this part of the code to understand what role that condition plays when evented pleg feature is enabled.

Reply:
Thanks @pacoxu. If we find this change actually fixes the issue, then I think we can add this explanation in comment (just above that if condition). WDYT?

Reply:
@harche makes sense. add more comments will be better.

Reply:
the feature gate is supposed to be removed once the feature graduates, we can not depend on the feature gate to act as a flag, am I misunderstanding this solution?

Reply:
@aojea we would need to make certain decisions based on whether PLEG in use is Evented or Generic. So we need some sort of boolean to distinguish that. 

Maybe instead of using FeatureGate, how about we introduce a kubelet config/flag, something like `--evented-pleg=true/false`?

Reply:
So instead of directly using `utilfeature.DefaultFeatureGate.Enabled(features.EventedPLEG)`, during the kubelet startup we could set a global boolean `EventedPleg` and use that throughout the code.

Reply:
@harche you know better than me for sure, I just wanted to say that you can not rely on the feature flag ,so do you plan to have two PLEG in the future or EventedPLEG will replace the existing?

Reply:
IMO, EventedPLEG is a better solution and we will use EventedPLEG by default in the future.

And logically, kubelet will fall back to use Generic PLEG if EventedPLEG didn't work or the container runtime didn't support it.

Reply:
this is why we have the KEPs and the PRRs sections, to discuss these things before implementing it, if the goal is to support to PLEG modes this should be clear https://github.com/kubernetes/enhancements/tree/master/keps/sig-node/3386-kubelet-evented-pleg#risks-and-mitigations

A quick skim of the KEP seems to indicate the assumption was to use the featureGate as flag, but that is incorrect, feature gates need to be removed on GA is reached. If we want to support multiple kubelet PLEG we can indicate with an enum for `PLEGMode` defaulting to one and switching to the other on GA, per exmaple, or with a bool, ... but this is something you should decide in SIG Node, I´m just throwing ideas